cmds/core/dirname/dirname.go

```diff
@@ -9,17 +9,25 @@ package main
 
 import (
 	"fmt"
+	"io"
 	"log"
 	"os"
 	"path/filepath"
 )
 
-func main() {
-	if len(os.Args) < 2 {
-		log.Fatalf("dirname: missing operand")
+func run(out io.Writer, args []string) error {
+	if len(args) < 1 {
+		return fmt.Errorf("missing operand")
```

Comment:
might as well make this a variable
var ErrNoArg = errors.New("missing operand")
then use the errors package in your tests.

Reply:
see bfe3c18